absl/strings/escaping_test.cc

```diff
@@ -278,6 +278,98 @@ TEST_F(CUnescapeTest, UnescapesMultipleUnicodeNulls) {
                    "\0", 5), result_string_);
 }
 
+TEST(CUnescape, NonNullTerminatedOctal1) {
+  absl::string_view reference("\\111", 2);
+
+  std::string unescaped_str;
+  EXPECT_TRUE(absl::CUnescape(reference, &unescaped_str));
+
+  EXPECT_EQ(unescaped_str, "\1");
+}
+
+TEST(CUnescape, NonNullTerminatedOctal2) {
+  absl::string_view reference("\\1z1", 3);
+
+  std::string unescaped_str;
+  EXPECT_TRUE(absl::CUnescape(reference, &unescaped_str));
+
+  EXPECT_EQ(unescaped_str, "\1z");
+}
+
+TEST(CUnescape, NonNullTerminatedHex) {
+  absl::string_view reference("\\x11", 3);
+
+  std::string unescaped_str;
+  EXPECT_TRUE(absl::CUnescape(reference, &unescaped_str));
+
+  EXPECT_EQ(unescaped_str, "\x1");
+}
+
+TEST(CUnescape, NonNullTerminatedInvalidHex) {
+  absl::string_view reference("\\x1", 2);
+
+  std::string unescaped_str;
+  EXPECT_FALSE(absl::CUnescape(reference, &unescaped_str));
+}
+
+TEST(CUnescape, NonNullTerminatedInvalidUtf1) {
```

Comment:
I prefer test names to describe how they differ rather than having numbers at the end where possible.  How about calling this `ShortUnicodeUnescapingDoesNotOverrunBuffer`?  (I hope that isn't so long it causes a line wrap.)